core/src/main/scala/za/co/absa/spline/harvester/builder/OperationNodeBuilder.scala

```diff
@@ -35,13 +34,21 @@ trait OperationNodeBuilder {
   def +=(childBuilder: OperationNodeBuilder): Unit = childBuilders :+= childBuilder
 
   protected def componentCreatorFactory: ComponentCreatorFactory
-  protected lazy val outputSchema: Schema = operation.output.map(componentCreatorFactory.attributeConverter.convert(_).id)
+  private lazy val attributeConverter =
+    if (isTerminal)
+      componentCreatorFactory.inputAttributeConverter
+    else
+      componentCreatorFactory.expressionConverter
+
+  private def convert(att: SparAttribute): String = attributeConverter.convert((att, id)).id
+  protected lazy val outputAttributes: OutputAttIds = operation.output.map(convert).toList
 
-  protected def childIds: Seq[Int] = childBuilders.map(_.id)
-  protected def childOutputSchemas: Seq[Schema] = childBuilders.map(_.outputSchema)
+  protected def childIds: List[OperationId] = childBuilders.map(_.id).toList
```

Comment:
Here we can use `Seq` as well with no extra efforts. The `.toList` call will just move from here to the place of usage.